### ISSUE TITLE
Broken package on pypi 

### ISSUE THREAD

**Issue by quom** [pytorch/pytorch#97153]
### 🐛 Describe the bug

Since the poetry upgrade to use a new installer (https://github.com/python-poetry/poetry/pull/7671) package hashes/size are checked against their metadata. This was giving the following error in `poetry install`
```
  ["In /home/runner/.cache/pypoetry/artifacts/7a/b0/7a/fec66e2cb3ecea3a443848620e6749d35cc3befc352ffc2be0d48b4397/torch-1.11.0-cp38-cp38-manylinux1_x86_64.whl, hash / size of torch-1.11.0.dist-info/METADATA didn't match RECORD"]

  at ~/.local/venv/lib/python3.8/site-packages/installer/sources.py:289 in validate_record
      285│                         f"In {self._zipfile.filename}, hash / size of {item.filename} didn't match RECORD"
      286│                     )
      287│ 
      288│         if issues:
    → 289│             raise _WheelFileValidationError(issues)
      290│ 
      291│     def get_contents(self) -> Iterator[WheelContentElement]:
      292│         """Sequential access to all contents of the wheel (including dist-info files).
      293│ 
```

After downloading the wheel manually and verifying I can see that the METADATA file has the wrong metadata in RECORD:
```
torch-1.11.0.dist-info/METADATA,sha256=x2b7fcuKitL9bgjcoY_QoDn6F9ocwlkDxhK5tbku_ag,24383
```
SHA246:
```sh
$ python3 -c "import hashlib; import base64; print(base64.b64encode(hashlib.sha256(open('METADATA', 'rb').read()).digest()))"

b'shdSa8B3YFfHAfuF0k9niOKJDeQ8VfqWHKlsUWktK0s='
```
File size in bytes:
```
24377 METADATA
```

I am not sure whether it is possible to retrospectively fix this issue but it may by symptomatic of an underlying issue with building packages.

### Versions

So far torch 1.11.0 manylinux x86_64 has been identified as having this problem: https://pypi.org/project/torch/1.11.0/#copy-hash-modal-d41167e9-e8bc-42d0-a2a7-fea81401c4a6

cc @ezyang @gchanan @zou3519 @seemethere @malfet

**Comment by rahul-ngt** [comment]
I am seeing the same issue with https://pypi.org/project/torch/1.8.1/#copy-hash-modal-8530d660-7cf3-435e-9c18-a677afc1f46f

**Comment by atalman** [comment]
I think this issue was adressed here: https://github.com/pytorch/builder/pull/1164

**Comment by malfet** [comment]
Is there an established mechanism to fix metadata for the older releases? 
But according to the @atalman it should be fixed for 1.13.x+. By the way, is there a standard script that validates the METADATA before uploading?

**Comment by Skylion007** [comment]
@malfet This package does various checks for wheels, not sure it would catch this specific bug though: https://pypi.org/project/check-wheel-contents/

**Comment by atalman** [comment]
closing this since fixed in 2.0
